test/SlowTests/Voron/Storage/SyncFlushTimingTest.cs

```diff
@@ -0,0 +1,218 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using FastTests.Voron;
+using Sparrow;
+using Voron;
+using Voron.Impl.Journal;
+using Xunit;
+
+namespace SlowTests.Voron.Storage
+{
+  public class SyncFlushTimingTest : StorageTest
+  {
+    [Fact]
+    public void CanSyncWhileFlushWaiteToWriteTransaction()
+    {
+      var doKeepNotIdle = true;
+      var keepGlobalFlushingBehaviorNotIdle = new Thread(() =>
+      {
+        while (doKeepNotIdle)
+        {
+          GlobalFlushingBehavior.GlobalFlusher.Value.MaybeFlushEnvironment(Env);
+          Thread.Sleep(1000);
+        }
+      });
+      keepGlobalFlushingBehaviorNotIdle.Start();
+
+      for (var i = 0; i < 100; i++)
+      {
+        using (var tx = Env.WriteTransaction())
+        {
+          var tree = tx.CreateTree("foo");
+          tree.Add("items/" + i, StreamFor("values/" + i));
+          tx.Commit();
+        }
+      }
+
+      Env.Options.ManualFlushing = true;
+      Env.FlushLogToDataFile();
+
+      for (var i = 0; i < 100; i++)
+      {
+        using (var tx = Env.WriteTransaction())
+        {
+          var tree = tx.CreateTree("foo");
+          tree.Add("items/" + i, StreamFor("values/" + i));
+          tx.Commit();
+        }
+      }
+
+      var sync = new Thread(() =>
```

Comment:
You are spawning a lot of threads, with no error handling or waiting for them to complete